drivers/ds18/ds18.c

```diff
@@ -184,8 +184,8 @@ int ds18_read(const ds18_t *dev, int16_t *temperature)
 
     DEBUG("[DS18] Received byte: 0x%02x\n", b2);
 
-    int32_t measurement = ((int16_t)(b2 << 8 | b1) * 625);
-    *temperature = (int16_t)(measurement / 100);
+    int16_t measurement = (int16_t)(b2 << 8 | b1);
```

Comment:
Since you are casting to `int32_t` anyway (and that being it's only use) you might as well use `int32_t` here and drop the cast - it should not make a difference for the generated code.